packages/sdk-api/src/api/services/RoleService.ts

```diff
@@ -79,7 +81,63 @@ export class RoleService extends CommonService {
     return __request<RoleResponse>(
       {
         method: 'GET',
-        url: `/roles/${roleId}`,
+        url: `/${ROLES_ENDPOINT}/${roleId}`,
+      },
+      this.openApi
+    );
+  }
+
+  /**
+   * Update role
+   *
+   * @param roleId string Role identifier
+   * @param body UpdateRoleRequest
+   *
+   * @returns RoleResponse Successful Response
+   *
+   * @throws ApiError
+   */
+  public update(
+    roleId: string,
+    body: UpdateRoleRequest
+  ): CancelablePromise<RoleResponse> {
+    return __request<RoleResponse>(
+      {
+        method: 'PATCH',
+        url: `/${ROLES_ENDPOINT}/${roleId}`,
+        body,
+        errors: {
+          400: `Bad Request`,
+          405: `Method Not Allowed`,
+          422: `Validation Error`,
+          500: `Internal Server Error`,
+        },
+      },
+      this.openApi
+    );
+  }
+
+  /**
+   * Create role
+   *
+   * @param body CreateRoleRequest
+   *
+   * @returns RoleResponse Successful Response
+   *
+   * @throws ApiError
+   */
+  public create(body: CreateRoleRequest): CancelablePromise<RoleResponse> {
+    return __request<RoleResponse>(
+      {
+        method: 'POST',
+        url: `/${ROLES_ENDPOINT}`,
+        body,
+        errors: {
+          400: `Bad Request`,
+          405: `Method Not Allowed`,
+          422: `Validation Error`,
+          500: `Internal Server Error`,
```

Comment:
Don't we want to create a separate object fort those errors collections to follow more DRY principle and avoid duplication?
```ts
const ERRORS = {
  400: `Bad Request`,
  405: `Method Not Allowed`,
  422: `Validation Error`,
  500: `Internal Server Error`,
};
```

```ts
return __request<RoleResponse>(
  {
    method: 'PATCH',
    url: `/${ROLES_ENDPOINT}/${roleId}`,
    body,
    errors: ERRORS,
  },
      this.openApi
    );
```

Reply:
That is a fair point, but the list of possible errors could be different from one endpoint to another. Even in the RoleService, there are different lists of errors for the `getList` method and `getDetail`.

Also, these lists are autogenerated from OpenApi schema with small by-hand fixes. Shortly, we will remove them with the QRaft, Aleksandr already started this work.

Reply:
I'm thinking maybe using a helper function to generate specific error objects as needed on fly if a good idea, somethink like this:

```ts
/**
 * Helper function to get error messages based on provided error codes.
 * This helps in adhering to the DRY principle by avoiding duplication of error messages across different endpoints.
 *
 * @param {Array<number>} errorCodes - Array of error codes.
 * @returns {Object} - Object containing the error messages for the provided error codes.
 *
 * Example usage:
 * const errorCodes = [400, 405, 422, 500];
 * const errors = getErrorMessages(errorCodes);
 * 
 * return __request<RoleResponse>(
 *   {
 *     method: 'PATCH',
 *     url: `/${ROLES_ENDPOINT}/${roleId}`,
 *     body,
 *     errors: errors,
 *   },
 *   this.openApi
 * );
 */
function getErrorMessages(errorCodes) {
  const errors = {};
  errorCodes.forEach(code => {
    if (ALL_ERRORS[code]) {
      errors[code] = ALL_ERRORS[code];
    }
  });
  return errors;
}
```

Reply:
@vkuprin, this one piece of code - is our legacy API Client. It was autogenerated previously. Since something went wrong with it, we were updating it's codebase sometime without any improvements.

This is the moment when Qraft enters the stage.